src/traces/streamtube/calc.js

```diff
@@ -165,32 +208,43 @@ module.exports = function calc(gd, trace) {
         empty();
     }
 
-    for(i = 0; i < slen; i++) {
-        var sx = startx[i];
-        xMax = Math.max(xMax, sx);
-        xMin = Math.min(xMin, sx);
+    return {
+        xMin: xMin,
+        yMin: yMin,
+        zMin: zMin,
+        xMax: xMax,
+        yMax: yMax,
+        zMax: zMax,
+        Xs: Xs,
+        Ys: Ys,
+        Zs: Zs,
+        len: len,
+        fill: gridFill
+    };
+}
 
-        var sy = starty[i];
-        yMax = Math.max(yMax, sy);
-        yMin = Math.min(yMin, sy);
+function distinctVals(col) {
+    return Lib.distinctVals(filter(col)).vals;
+}
```

Comment:
I think this should be

```js
function distinctVals(col, len) {
    return Lib.distinctVals(filter(col, len)).vals;
}
```

Reply:
Good eye.
Done in https://github.com/plotly/plotly.js/pull/4431/commits/b71677a1255b52cab4dae423de4d501a6f0b4a81.